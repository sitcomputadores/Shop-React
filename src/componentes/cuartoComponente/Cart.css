.containerImg{
    text-align: center;
}

.imgText{
    width: 10rem;
    height: 10rem;
}

.nameText{
    color: black;
    margin-bottom: 2rem;
}