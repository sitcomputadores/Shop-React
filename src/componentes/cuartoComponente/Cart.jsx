import React, { useState } from 'react';
import './Cart.css';

export const Cart = () => {

    const [texto, setTexto] = useState('');

    const onClickImg = () => {
        setTexto('Luis Fernando Heredia Forero')
    }

    return (
        <>
            <div className='containerImg'>
                <img onClick={ onClickImg } className='imgText' src="https://img.freepik.com/vector-gratis/diseno-logotipo-tienda-ropa-diseno-plano_23-2149496415.jpg?size=338&ext=jpg&ga=GA1.1.2116175301.1701043200&semt=ais" alt="" />
                <h1 className='nameText'>{texto}</h1>
            </div>
        </>
    )
}