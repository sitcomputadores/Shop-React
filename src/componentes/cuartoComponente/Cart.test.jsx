import { render, screen, fireEvent } from "@testing-library/react";
import { Cart } from "./Cart";

jest.mock('./Cart.css', () => '')
describe('Pruebas a componente', () => {
    test('Renderizar componente', () => {
        const { container } = render(<Cart />);
        expect(container).toMatchSnapshot();
    });

    test('Traer imagen', () => {
        render(<Cart />);
        const { src } = screen.getByRole('img');
        expect(src).toEqual('https://img.freepik.com/vector-gratis/diseno-logotipo-tienda-ropa-diseno-plano_23-2149496415.jpg?size=338&ext=jpg&ga=GA1.1.2116175301.1701043200&semt=ais');
    });

    test('Validar texto null', () => {
        render(<Cart />);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain('');
    });

    test('Identificar texto', () => {
        render(<Cart />);
        fireEvent.click(screen.getByRole('img'));
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain('Luis');
    });
})