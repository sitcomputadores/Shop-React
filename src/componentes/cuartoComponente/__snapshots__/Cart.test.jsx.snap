// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente Renderizar componente 1`] = `
<div>
  <div
    class="containerImg"
  >
    <img
      alt=""
      class="imgText"
      src="https://img.freepik.com/vector-gratis/diseno-logotipo-tienda-ropa-diseno-plano_23-2149496415.jpg?size=338&ext=jpg&ga=GA1.1.2116175301.1701043200&semt=ais"
    />
    <h1
      class="nameText"
    />
  </div>
</div>
`;
